ferrocene/tools/self-test/src/linkers.rs

```diff
@@ -430,4 +774,235 @@ mod tests {
+            }
+            Err(e) => {
+                // Correct
+                assert!(compiler_args.is_empty());
+            }
+        }
+    }
+
+    #[test]
+    fn test_linker_args_unknown_missing_long_arg() {
+        let linker_args = ["--plugin"];
+        let mut compiler_args = Vec::new();
+        match linker_args_ok(
+            "x86_64-unknown-linux-gnu",
+            linker_args.iter().cloned(),
+            &mut compiler_args,
+        ) {
+            Ok(_) => {
+                panic!("Unexpected acceptance processing {:?}", linker_args);
+            }
+            Err(e) => {
+                // Correct
+                assert!(compiler_args.is_empty());
+            }
+        }
+    }
+
+    #[test]
+    fn test_rationalise_linker_args() {
+        let args = [
+            "-plugin",
+            "/usr/lib/gcc-cross/aarch64-linux-gnu/12/liblto_plugin.so",
+            "-plugin-opt=/usr/lib/gcc-cross/aarch64-linux-gnu/12/lto-wrapper",
+            "-plugin-opt=-fresolution=/tmp/cc9GsGIt.res",
+            "-plugin-opt=-pass-through=-lgcc",
+            "-plugin-opt=-pass-through=-lgcc_s",
+            "-plugin-opt=-pass-through=-lc",
+            "-plugin-opt=-pass-through=-lgcc",
+            "-plugin-opt=-pass-through=-lgcc_s",
+            "--sysroot=/",
+            "--build-id",
+            "--eh-frame-hdr",
+            "--hash-style=gnu",
+            "--as-needed",
+            "-dynamic-linker",
+            "/lib/ld-linux-aarch64.so.1",
+            "-X",
+            "-EL",
+            "-maarch64linux",
+            "--fix-cortex-a53-843419",
+            "-pie",
+            "-z",
+            "now",
+            "-z",
+            "relro",
+            "-o",
+            "/tmp/.tmpoNgAJE/output.bin",
+            "/usr/lib/gcc-cross/aarch64-linux-gnu/12/../../../../aarch64-linux-gnu/lib/../lib/Scrt1.o",
+            "/usr/lib/gcc-cross/aarch64-linux-gnu/12/../../../../aarch64-linux-gnu/lib/../lib/crti.o",
+            "/usr/lib/gcc-cross/aarch64-linux-gnu/12/crtbeginS.o",
+            "-L/tmp/.tmpoNgAJE",
+            "-L/usr/lib/gcc-cross/aarch64-linux-gnu/12",
+            "-L/usr/lib/gcc-cross/aarch64-linux-gnu/12/../../../../aarch64-linux-gnu/lib/../lib",
+            "-L/lib/aarch64-linux-gnu",
+            "-L/lib/../lib",
+            "-L/usr/lib/aarch64-linux-gnu",
+            "-L/usr/lib/../lib",
+            "-L/usr/lib/gcc-cross/aarch64-linux-gnu/12/../../../../aarch64-linux-gnu/lib",
+            "/tmp/cc81JpAZ.o",
+            "-lgcc",
+            "--push-state",
+            "--as-needed",
+            "-lgcc_s",
+            "--pop-state",
+            "-lc",
+            "-lgcc",
+            "--as-needed",
+            "-lgcc_s",
+            "--no-as-needed",
+            "/usr/lib/gcc-cross/aarch64-linux-gnu/12/crtendS.o",
+            "/usr/lib/gcc-cross/aarch64-linux-gnu/12/../../../../aarch64-linux-gnu/lib/../lib/crtn.o",
+        ];
+        let expected = vec![
```

Comment:
So ...  a file foobar1.txt containing the text version of the linker arguments and a file called ... foobar1.json which contains a JSON serialised version of the `Vec<LinkerArgs>`?

Reply:
Yes, or even keeping the list of flags in the Rust source code and delegating to insta just the outcome of the parsing. Mentioning insta because it automatically shows a diff when a test fails, and has a command to accept all changes automatically.

Reply:
Wait, no, have read the docs now :)

Reply:
Added. I like it!